packages/js/components/src/experimental-select-control/combo-box.tsx

```diff
@@ -1,30 +1,60 @@
 /**
  * External dependencies
  */
-import { createElement } from 'react';
+import { createElement, MouseEvent, useRef } from 'react';
 import { Icon, search } from '@wordpress/icons';
+// import { MouseEvent } from 'react';
 
 /**
  * Internal dependencies
  */
 import { Props } from './types';
 
 type ComboBoxProps = {
+	children?: JSX.Element | JSX.Element[] | null;
 	comboBoxProps: Props;
 	inputProps: Props;
+	inputRef?: React.RefObject< HTMLInputElement > | null;
 };
 
-export const ComboBox = ( { comboBoxProps, inputProps }: ComboBoxProps ) => {
+export const ComboBox = ( {
+	children,
+	comboBoxProps,
+	inputProps,
+	inputRef = null,
+}: ComboBoxProps ) => {
+	const maybeFocusInput = ( event: MouseEvent< HTMLDivElement > ) => {
+		if ( ! inputRef || ! inputRef.current ) {
+			return;
+		}
+
+		event.preventDefault();
+
+		if ( document.activeElement !== inputRef.current ) {
+			inputRef.current.focus();
+			event.stopPropagation();
+		}
+	};
+
 	return (
+		// Disable reason: The click event is purely for accidental clicks around the input.
+		// Keyboard users are still able to tab to and interact with elements in the combobox.
+		/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
 		<div
-			{ ...comboBoxProps }
-			className="woocommerce-experimental-select-control__combox-box"
+			className="woocommerce-experimental-select-control__combo-box-wrapper"
+			onMouseDown={ maybeFocusInput }
 		>
-			<input { ...inputProps } />
-			<Icon
-				className="woocommerce-experimental-select-control__combox-box-icon"
-				icon={ search }
-			/>
+			{ children }
+			<div
+				{ ...comboBoxProps }
+				className="woocommerce-experimental-select-control__combox-box"
+			>
+				<input { ...inputProps } />
+				<Icon
```

Comment:
I am not entire opposed to pushing the icon down with the `Input` field, but would it look better if it stay's centred? 
Or just focused to the top.
I guess having this icon inline with the input, does make it easier for the user to find the input field.

Mostly thinking out loud on this one, I am entirely fine leaving it as is.

Reply:
I was wondering the same.  They both feel weird to me, but I liked how the icon helped identify where the input was faster.  @jarekmorawski do you have any opinion on this?

Reply:
Not a very strong one, but I'd probably keep it centered for simplicity's sake.